src/goose/brokers/redis/commands.clj

```diff
@@ -183,6 +199,22 @@
                     [next-cursor elements]))]
     (scan-seq conn scan-fn list-key size)))
 
+(defn range-from-front
+  "Fetches range of elements from front where start and stop are relative to the front of list"
+  [conn list-key start stop]
+  (let [size (list-size conn list-key)
+        min-start (if (>= (- size stop 1) 0) (- size stop 1) 0)
+        min-stop (if (>= (- size start 1) 0) (- size start 1) 0)]
+    (reverse (wcar* conn (car/lrange list-key min-start min-stop)))))
+
+;;Approach 2
```

Comment:
note: This is another alternative for the same function. It uses negative indexing to extract entries from the queue.

The approach 1 is more explicit in the start and stop values, approach 2 is simpler and leverages the use of negative indexes in redis.